spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -1112,7 +1152,43 @@ else if (replyTo instanceof TemporaryQueue || replyTo instanceof TemporaryTopic)
 		return reply;
 	}
 
-	private void sendRequestMessage(javax.jms.Message jmsRequest, MessageProducer messageProducer, int priority) throws JMSException {
+	private SettableListenableFuture<Message<?>> createFuture(final String correlationId) {
+		SettableListenableFuture<Message<?>> future = new SettableListenableFuture<Message<?>>();
+		this.futures.put(correlationId, future);
+		if (this.receiveTimeout > 0) {
+			getTaskScheduler().schedule(new Runnable() {
+
+				@Override
+				public void run() {
+					expire(correlationId);
+				}
+			}, new Date(System.currentTimeMillis() + this.receiveTimeout));
+		}
+		return future;
+	}
+
+	private void expire(String correlationId) {
+		final SettableListenableFuture<Message<?>> future = JmsOutboundGateway.this.futures
+				.remove(correlationId);
```

Comment:
The `JmsOutboundGateway.` is redundant and the `remove` can be on the same line.
Doesn't hurt, of course. Will fix on merge.